bgs/bgs.go

```diff
@@ -406,6 +407,39 @@ func prometheusHandler() http.Handler {
 	return exporter
 }
 
+func (s *BGS) domainIsBanned(ctx context.Context, host string) (bool, error) {
```

Comment:
A doc comment explaining that this works by checking the full domain, then all the parent domains, up to and including the TLD, would be helpful.